test/deform/linearized_deform_test.py

```diff
@@ -0,0 +1,302 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Test operators of linearized deformations"""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+
+# External
+import pytest
+import numpy as np
+
+# Internal
+import odl
+from odl.util.testutils import all_equal
+
+
+def hat_function(x, **kwargs):
+    """Hat function with support in square with side 2r."""
+    r = kwargs.pop('r', 0.5)
+    retval = (r - abs(x[0])) * (abs(x[0]) < r)
+    for xi in x[1:]:
+        retval = retval * (r - abs(xi)) * (abs(xi) < r)
+    return retval
+
+
+def hat_function_grad0(x, **kwargs):
+    """First component of gradient of hat function with support in square
+    with side 2r."""
+    r = kwargs.pop('r', 0.5)
+    tmp = np.where(abs(x[0]) >= 1e-6, -x[0] / abs(x[0]) * (r - abs(x[1])), 0)
+    return tmp * (abs(x[0]) < r) * (abs(x[1]) < r)
+
+
+def hat_function_grad1(x, **kwargs):
+    """2nd component of gradient of hat function with support in square
+    with side 2r."""
+    r = kwargs.pop('r', 0.5)
+    tmp = np.where(abs(x[1]) >= 1e-6, -x[1] / abs(x[1]) * (r - abs(x[0])), 0)
+    return tmp * (abs(x[0]) < r) * (abs(x[1]) < r)
+
+
+def disp_field0(x):
+    """First component of analytic displacement field (x,y) -> eps * (x+y)."""
+    eps = 0.25
+    return eps * (x[0] + x[1])
+
+
+def disp_field1(x):
+    """Second component of analytic displacement field (x,y) -> eps * 3xy."""
+    eps = 0.25
+    return eps * 3 * x[0] * x[1]
+
+
+def deform_hat(x, **kwargs):
+    """Analytic deformed hat function with support in square with side r.
+    The analytic displacement field (x,y) -> eps * (x+y, 3xy)."""
+    r = kwargs.pop('r', 0.5)
+    eps = 0.25
+    x0 = x[0] + eps * (x[0] + x[1])
```

Comment:
Should call the underlying function for clarity and less redundancy

x0 = x[0] + disp_field0(x)
x1 = x[1] + disp_field1(x)